includes/class-sensei-course.php

```diff
@@ -1418,7 +1418,13 @@ public function load_user_courses_content( $user = false ) {
 					/**
 					 * documented in class-sensei-course.php the_course_action_buttons function
 					 */
-					$show_delete_course_button = apply_filters( 'sensei_show_delete_course_button', false );
+					$show_delete_course_button = apply_filters_deprecated(
```

Comment:
I'd be OK with deprecating this, but removing it in 2.1.0 seems far too quick. I'm not sure we should commit ourselves to removal in a particular release at all, so I'd prefer we don't specify a version number at all.

Also, if we get any backlash from this deprecation (doubtful, but possible), we may have to reconsider.

Reply:
> removing it in 2.1.0 seems far too quick.

I'm ok with bumping it to a later version. Maybe 2.2.0 to give it two minor releases as a deprecation? (2.0 and 2.1)

> I'm not sure we should commit ourselves to removal in a particular release

I disagree, and I think we need to get out of this habit 🙂 

If we don't commit to a release, there's a good chance we'll never remove the deprecated code (as seen with all the rest of the deprecated or legacy code that has been in Sensei for years). Deprecating code seems pointless if we're not actually going to remove it. The point is to reduce technical debt, and give us the ability to make needed changes and stop maintaining legacy code that we're worried someone might still be using.

Targetting a release with our deprecations means that we set expectations, and give developers and users a timeframe to update their code or implement the workaround. In my opinion, this is much more beneficial and responsible than leaving deprecated code in the codebase indefinitely.

For this particular feature, if we're not comfortable committing to removing it in an upcoming release, then let's not deprecate it, and make another decision on what to do here.

Reply:
@donnapep As per the P2 discussion, I changed the deprecation to November 2019. I also added the issue for the removal into a milestone - https://github.com/Automattic/sensei/milestone/56